sound/soc/sof/mediatek/mt8195/mt8195.c

```diff
@@ -316,6 +316,19 @@ static int mt8195_dsp_suspend(struct snd_sof_dev *sdev, u32 target_state)
 {
 	struct platform_device *pdev = container_of(sdev->dev, struct platform_device, dev);
 	int ret;
+	u32 reset_sw, dbg_pc;
+
+	/* wait dsp enter idle, timeout is 1 second */
+	ret = snd_sof_dsp_read_poll_timeout(sdev, DSP_REG_BAR,
+					    DSP_RESET_SW, reset_sw,
+					    ((reset_sw & ADSP_PWAIT) == ADSP_PWAIT),
+					    SUSPEND_DSP_IDLE_POLL_INTERVAL_US,
+					    SUSPEND_DSP_IDLE_TIMEOUT_US);
+	if (ret < 0) {
+		dbg_pc = snd_sof_dsp_read(sdev, DSP_REG_BAR, DSP_PDEBUGPC);
+		dev_err(sdev->dev, "dsp not idle : swrest %#x, pc %#x, ret %d\n",
+			reset_sw, dbg_pc, ret);
+	}
```

Comment:
@yaochunhung if you are going to power off anyway even if the DSP is not idle after the timeout, the wait seems meaningless.

Reply:
@ranj063 basically in normal flow I think power off should be done after dsp enter idle to avoid unexpected (random) behavior. But I don't want to stop entire system sleeping due to DSP is abnormal.  I can try to add delay in DSP firmware after receiving suspend command to check if it works fine for next resume but I think it is safer to keep checking even this experiment works. Thanks.
```c
Test code :
void ipc_platform_complete_cmd(struct ipc *ipc)
{
   	while (ipc->pm_prepare_D3) {
                wait_delay <= add delay here.
		asm volatile("waiti 15");
         }
}
```

Reply:
> @ranj063 basically in normal flow I think power off should be done after dsp enter idle to avoid unexpected (random) behavior. But I don't want to stop entire system sleeping due to DSP is abnormal. I can try to add delay in DSP firmware after receiving suspend command to check if it works fine for next resume but I think it is safer to keep checking even this experiment works. Thanks.

@yaochunhung Can the system enter suspend if you dont power off the DSP? I think its better to fail early and prevent system suspend in this case so you know the culprit no?

Reply:
@ranj063 DSP still be powered off at kernel side even if DSP doesn't enter idle(DSP firmware only let DSP enter idle, not power off). Basically I saw DSP will enter idle immediately per my testing(about 10000 times stress) under this checking flow. Thanks.

Reply:
ok I think the dev_err needs to be demoted to dev_warn in this case saying that "DSP not idle, powering off anyway" or somethign similar

Reply:
OK. I will modify it. thanks.